### PR TITLE
Clean up old Rust gRPC compile code that used the 'grpc' crate

### PR DESCRIPTION
**Pull request by alexjpwalker**
## What is the goal of this PR?

We cleaned up old Rust gRPC compile code that was using the `grpc` crate, as we are now using the `tonic` crate instead.

## What are the changes implemented in this PR?

We successfully migrated `typedb-client-rust` from using the `grpc` crate to the more actively-maintained `tonic` crate, and from the `protobuf` crate (used by `grpc`) to the `prost` crate used by `tonic`.

So, we can now freely delete the old gRPC compiler code using the `grpc` crate from `typedb-protocol`.

We've also integrated the Rust IDE sync tool into `typedb-protocol`, which is needed to provide IntelliSense in our Rust sources.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -36,8 +36,13 @@ bazel-*
 # Compiled files #
```
Comment (alexjpwalker) [inline .gitignore:36]: We've integrated the Rust IDE sync tool into `typedb-protocol`, which can result in the generation of `target/`, `Cargo.toml` and `Cargo.lock`.

File: BUILD [hunk 2]
```diff
@@ -59,6 +59,7 @@ exports_files(["README.md"])
 filegroup(
     name = "ci",
     data = [
+        "@vaticle_dependencies//ide/rust:sync",
```
Comment (alexjpwalker) [inline BUILD:62]: Add the Rust IDE sync tool as a checked target.

File: WORKSPACE [hunk 3]
```diff
@@ -76,10 +76,6 @@ java_grpc_compile()
 load("@stackb_rules_proto//node:deps.bzl", "node_grpc_compile")
 node_grpc_compile()
 
-load("@rules_rust//proto:repositories.bzl", "rust_proto_repositories")
-rust_proto_repositories()
-register_toolchains("//grpc/rust:proto_toolchain")
```
Comment (alexjpwalker) [inline WORKSPACE:81]: This was required by the old `grpc` crate code.

File: dependencies/vaticle/repositories.bzl [hunk 4]
```diff
@@ -21,5 +21,5 @@ def vaticle_dependencies():
     git_repository(
         name = "vaticle_dependencies",
         remote = "https://github.com/vaticle/dependencies",
-        commit = "4464b506ca469f37d3b696fb2f1eda34061cdaa1", # sync-marker: do not remove this comment, this is used for sync-dependencies by @vaticle_dependencies
+        commit = "5be6d949ca1e04e4179ea6acb3432be713b9dfb8", # sync-marker: do not remove this comment, this is used for sync-dependencies by @vaticle_dependencies
```
Comment (alexjpwalker) [inline dependencies/vaticle/repositories.bzl:24]: The old crates that became redundant through this PR have been deleted in this version of `dependencies`.

**Review by haikalpribadi — APPROVED** [review]

**Pull request merged**